src/components/CreditCardForm.test.js

```diff
@@ -0,0 +1,114 @@
+      const message = await screen.findByText("Visa");
+      expect(message).toBeInTheDocument();
+    });
+  });
+  it("cardNumber input - check card type -> Visa(13 digits)", async () => {
+    render(<CreditCardForm />);
+    const fieldCardNumber = await screen.findByRole("textbox", {
+      name: /cardNumber/i,
+    });
+    userEvent.type(fieldCardNumber, "4123456789123");
+    const sendButton = await screen.findByRole("button", {
+      name: "send",
+    });
+    userEvent.click(sendButton);
+
+    await waitFor(async () => {
+      const message = await screen.findByText("Visa");
+      expect(message).toBeInTheDocument();
+    });
+  });
+  it("cardNumber input - check card type -> MasterCard", async () => {
+    render(<CreditCardForm />);
+    const fieldCardNumber = await screen.findByRole("textbox", {
+      name: /cardNumber/i,
+    });
+    userEvent.type(fieldCardNumber, "5123456789123123");
+    const sendButton = await screen.findByRole("button", {
+      name: "send",
+    });
+    userEvent.click(sendButton);
+
+    await waitFor(async () => {
+      const message = await screen.findByText("MasterCard");
+      expect(message).toBeInTheDocument();
+    });
+  });
+  it("cardNumber input - check card type -> AmericanExpress", async () => {
+    render(<CreditCardForm />);
+    const fieldCardNumber = await screen.findByRole("textbox", {
+      name: /cardNumber/i,
+    });
+    userEvent.type(fieldCardNumber, "312345678912312");
+    const sendButton = await screen.findByRole("button", {
+      name: "send",
+    });
+    userEvent.click(sendButton);
+
+    await waitFor(async () => {
+      const message = await screen.findByText("AmericanExpress");
+      expect(message).toBeInTheDocument();
+    });
+  });
+  it("cardNumber input - check card type -> wrong number", async () => {
+    render(<CreditCardForm />);
+    const fieldCardNumber = await screen.findByRole("textbox", {
+      name: /cardNumber/i,
+    });
+    userEvent.type(fieldCardNumber, "112345678912312");
+    const sendButton = await screen.findByRole("button", {
+      name: "send",
+    });
+    userEvent.click(sendButton);
+
+    await waitFor(async () => {
+      const message = await screen.findByText(
+        "Card number do not match any type!"
+      );
+      expect(message).toBeInTheDocument();
+    });
+  });
+  it("cardNumber input - check number(wrong)", async () => {
+    render(<CreditCardForm />);
+    const fieldCardNumber = await screen.findByRole("textbox", {
+      name: /cardNumber/i,
+    });
+    userEvent.type(fieldCardNumber, "1111111111111111");
+
+    await waitFor(async () => {
+      const message = await screen.findByText("Wrong card number!");
+      expect(message).toBeInTheDocument();
+    });
+  });
+
+  // poniższy test nie przechodzi, ale pojawia się informacja, że coś jest nie tak w <LoginForm> component i nie za bardzo rozumiem, jaki to ma tutaj związek?
+  /*it("cardNumber input - check number(correct)", async () => {
```

Comment:
Nie zauważyłem błędu dot. `<LoginForm>`, ale jeśli mamy jakiś problem z testem to warto go uruchomić samodzielnie przez dodanie only tj. `it.only("cardNu`

PS. Podobnie możemy pominąć inne testy z innego pliku przez dodanie x tj. `xdescribe("<Auth>", () => {`